tests/util/pool_tests.rs

```diff
@@ -97,3 +99,49 @@ pub fn test_filesystem_rename(paths: &[&Path]) {
 
     engine.teardown().unwrap();
 }
+
+/// Verify that destroy_filesystems actually deallocates the space
+/// from the thinpool, by attempting to reinstantiate it using the
+/// same thin id and verifying that it fails.
+pub fn test_thinpool_thindev_destroy(paths: &[&Path]) -> () {
+    let (mut pool, _) = StratPool::initialize("stratis_test_pool",
+                                              &DM::new().unwrap(),
+                                              paths,
+                                              Redundancy::NONE,
+                                              true)
+            .unwrap();
+    let &(_, fs_uuid) = pool.create_filesystems(&["stratis_test_filesystem"])
+        .unwrap()
+        .first()
+        .unwrap();
+
+    let fs_id = pool.get_strat_filesystem(&fs_uuid).unwrap().thin_id();
+
+    pool.destroy_filesystems(&[&fs_uuid]).unwrap();
+
+    let pool_uuid = *pool.uuid();
+
+    // Try to setup a thindev that has been destroyed
+    let dm = DM::new().unwrap();
+    let thindev = ThinDev::setup("stratis_test_thin_dev",
+                                 &dm,
+                                 pool.thinpooldev(),
+                                 fs_id,
+                                 Sectors(128u64));
+    assert!(thindev.is_err());
+    pool.teardown().unwrap();
+
+    // Check that destroyed fs is not present in MDV. If the record
```

Comment:
I've realized that relying on this behavior of ```setup()``` might make for too unstable a test. Perhaps it would be better just to take a look at the filesystem metadata on the mdv before tearing down the pool.